Flaky test: test_database_glue/test.py::test_show_tables_optimization

_Important: This issue was automatically generated and is used by CI for matching failures. DO NOT modify the body content. DO NOT remove labels._

Test name: test_database_glue/test.py::test_show_tables_optimization
Failure reason: Timeout
CI report: [Integration tests (amd_msan, 6/6)](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=106107&sha=eb69563e4011e51a18d940536a629fb975c4f148&name_0=PR&name_1=Integration%20tests%20%28amd_msan%2C%206%2F6%29)

Failing test history: [cidb](https://play.clickhouse.com#V0lUSAogICAgOTAgQVMgaW50ZXJ2YWxfZGF5cwpTRUxFQ1QKICAgIHRvU3RhcnRPZkRheShjaGVja19zdGFydF90aW1lKSBBUyBkYXksCiAgICBjb3VudCgpIEFTIGZhaWx1cmVzLAogICAgZ3JvdXBVbmlxQXJyYXkocHVsbF9yZXF1ZXN0X251bWJlcikgQVMgcHJzLAogICAgYW55KHJlcG9ydF91cmwpIEFTIHJlcG9ydF91cmwKRlJPTSBjaGVja3MKV0hFUkUgKG5vdygpIC0gdG9JbnRlcnZhbERheShpbnRlcnZhbF9kYXlzKSkgPD0gY2hlY2tfc3RhcnRfdGltZQogICAgQU5EIHRlc3RfbmFtZSA9ICd0ZXN0X2RhdGFiYXNlX2dsdWUvdGVzdC5weTo6dGVzdF9zaG93X3RhYmxlc19vcHRpbWl6YXRpb24nCiAgICAtLSBBTkQgY2hlY2tfbmFtZSA9ICdJbnRlZ3JhdGlvbiB0ZXN0cyAoYW1kX21zYW4sIDYvNiknCiAgICBBTkQgdGVzdF9zdGF0dXMgSU4gKCdGQUlMJywgJ0VSUk9SJykKICAgIEFORCAocHVsbF9yZXF1ZXN0X251bWJlciA9IDAgT1IgYmFzZV9yZWYgSU4gKCdtYXN0ZXInKSkKICAgIEFORCB0ZXN0X2NvbnRleHRfcmF3IExJS0UgJyVUaW1lb3V0JScKR1JPVVAgQlkgZGF5Ck9SREVSIEJZIGRheSBERVNDCg==)

Test output:
```
File: test_database_glue/test.py:903 - in test_show_tables_optimization
    node.query(f"SHOW TABLES FROM {CATALOG_NAME}", timeout=5)
File: helpers/cluster.py:5030 - in query
    return self.client.query(
File: helpers/client.py:40 - in wrap
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
File: helpers/client.py:80 - in query
    ).get_answer()
      ^^^^^^^^^^^^
File: helpers/client.py:264 - in get_answer
    raise QueryTimeoutExceedException("Client timed out!")
E   helpers.client.QueryTimeoutExceedException: Client timed out!
```

<!-- ch-version-info:start -->
### Version info
- Resolved by: #108144
- Merged into: `26.6.1.1144`
<!-- ch-version-info:end -->

--- a/tests/integration/test_database_glue/test.py
+++ b/tests/integration/test_database_glue/test.py
@@ -291,6 +291,21 @@ def started_cluster():
         cluster.shutdown()
 
 
+@pytest.fixture(autouse=True)
+def clean_catalog(started_cluster):
+    # All tests share one module-scoped Glue catalog. Listing operations such as
+    # SHOW TABLES enumerate every namespace and make one sequential Glue call per
+    # namespace, so leftover namespaces from earlier tests make these calls grow
+    # unbounded and eventually exceed query timeouts. Drop everything each test
+    # created so the catalog stays bounded to the running test.
+    yield
+    catalog = load_catalog_impl(started_cluster)
+    for namespace in catalog.list_namespaces():
+        for table in catalog.list_tables(namespace):
+            catalog.drop_table(table)
+        catalog.drop_namespace(namespace)
+
+
 def test_no_secrets_in_logs(started_cluster):
     node = started_cluster.instances["node1"]
 

